Automattic/simplenote-ios
# Spring Cleaning: New Note Info Sub-task

### Fix

This PR adds the new navigation button and implements the new Note Info UI behind it.

![ezgif-7-c9243cb9d8ce](https://user-images.githubusercontent.com/40267301/82117025-34864e00-97a0-11ea-80a8-e3e343517345.gif)

Sub-task of #741.

### Test Button

1. Select any note
2. Confirm the new "..." button is visible
3. Confirm the Note Info view appears correctly

### Test View

1. Confirm table ignores taps
2. Confirm note details are correct
3. Confirm bottom sheet responds to swipes
4. Confirm "x" button dismisses view

### Review

This PR is to merge changes to the `741-note-options` "feature" branch. Not into `develop`.

**Notes for** @SylvesterWilmott 

1. The "x" button doesn't match the system default. May need a new image to get this exactly right. Currently uses an existing image from the project.
2. Date formats do not match design spec. Currently using the localized system settings (ie: the user's settings). The Note Options class has a date formatter that matches the spec too, if that is preferred.
3. I'll need to know some min/max sizes for the sheet on the iPad. Right now it does some rough sizing but it's not what it should be.

Please let me know if there is anything that looks off!

**Notes for** @jleandroperez 

- I've set up the `741-note-options` branch as the "feature" branch.
- I went with a XIB here as storyboards where a bit harder to wrangle into a bottom sheet like this. No layout in code.
- The UITableView uses standard UITableViewCell for now as these match the design spec. We can always introduce a custom subclass if needed though.
- The Note Info view is fully localized in code.
- I set up the "x" button to use a block-based callback instead of a delegate.
- Local data is declared as optional. Not sure what the best practice is for Swift and would love to hear your thoughts on this.

I'm thinking`var note: Note?` vs `var note: Note!` vs ? If we went with `!` we could test and fatal in `viewDidLoad` but assume the values are set everywhere else, much like how IBOutlets are handled.

### Release

Initial release notes are already part of the `741-note-options` branch. Those will need to be reviewed before merging the feature branch back to `develop` but should be fine for now.
